### DISCUSSION TITLE
Are we interested if running an external program resulted in a crash / core dumped?

### DISCUSSION THREAD

**Discussion by sanderjo** [sabnzbd/sabnzbd#2824] [Ideas]
Triggered by the unrar problems https://github.com/sabnzbd/sabnzbd/issues/2823 ... which BTW is closed so not too important? Anyway:

Are we interested if running an external program resulted in a crash / core dumped?

subprocess.Popen() can tell via stderr if the program bailed out:

```
#!/usr/bin/env python3
from subprocess import Popen, PIPE

shell_command = "./divide_by 111"
print("\n", shell_command)
proc = Popen(shell_command, shell=True, stdout=PIPE, stderr=PIPE)
out, err = proc.communicate()
print("out",out)
print("err",err)
print("rc",proc.returncode)


shell_command = "./divide_by bbbb"
print("\n", shell_command)
proc = Popen(shell_command, shell=True, stdout=PIPE, stderr=PIPE)
out, err = proc.communicate()
print("out",out)
print("err",err)
print("rc",proc.returncode)

```

```
$ python3 poging2.py 

 ./divide_by 111
out b'Program name ./divide_by\nThe argument supplied is 111\ninteger is 111\ndivision gives 900\n'
err b''
rc 1

 ./divide_by bbbb
out b''
err b'Floating point exception (core dumped)\n'
rc 136

```
So if err is not empty, something bad happened. That might be relevant / revealing.

**Comment by Safihre** [comment]
Unfortunately, that won't work.
Regular applications can also write to stderr, and they do so. From memory it's for example 7zip that will write to stderr if the unpacking failed for regular reasons.
Therefore in all our POpen calls we redirect the stderr to stdout so we capture everything.

**Comment by sanderjo** [comment]
> Therefore in all our POpen calls we redirect the stderr to stdout so we capture everything.

Aha! So SAB can already detect '(core dumped)' in output ... ? 

(I've not been able to reproduce the unrar core dumped ... I haven't got that old x86-hardware lying around)
